packages/@aws-cdk/aws-logs-destinations/test/integ.lambda.ts

```diff
@@ -0,0 +1,46 @@
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as logs from '@aws-cdk/aws-logs';
+import * as cdk from '@aws-cdk/core';
+import { IntegTest } from '@aws-cdk/integ-tests';
+import { LambdaDestination } from '../lib';
+
+const app = new cdk.App();
+
+// Test with permission as dependency
+const stack = new cdk.Stack(app, 'aws-logs-destinations-lambda');
+
+const fn = new lambda.Function(stack, 'MyFunction', {
+  runtime: lambda.Runtime.NODEJS_16_X,
+  handler: 'index.handler',
+  code: lambda.Code.fromInline('exports.handler = function(event, ctx, cb) { return cb("hi", null); }'),
+});
+
+const logGroup = new logs.LogGroup(stack, 'LogGroup');
+new logs.SubscriptionFilter(stack, 'LogSubscriptionFilter', {
+  destination: new LambdaDestination(fn),
+  logGroup,
+  filterPattern: logs.FilterPattern.allEvents(),
+});
+
+// Test without permission as dependency
```

Comment:
Does this test work?

Also, can you add some assertions to make sure that the function is getting invoked? You could write a log to the log group and then somehow validate that the function gets invoked. Query the log group of the lambda function or something else.

Reply:
Yes this test works as expected, because it tests that the permission is not created :) I have added this info to the comment.

The feedback about the assertions is really good! I didn't thought about using them. I will add it like you suggested. Do you know how I can query the log group of the lambda in a good fashion? Because as far as I know the log sequence name will be generated in respect to the time it was created (with a date in the name I think). Would be great if you could give me a hint :)

Reply:
> Yes this test works as expected, because it tests that the permission is not created :) I have added this info to the comment.

I don't think we need this test. My point about it not working is that without the permission the function cannot be triggered. To truly test this "scenario" we would need to assert that it _did not_ work.

> Do you know how I can query the log group of the lambda in a good fashion?

You might be able to do something like this. Or you could even just assert that the log group for the lambda function exists since it doesn't get created until the lambda is invoked. You may run into an issue where the assertion runs before the log group exists (i'm working on allowing waiting) in which case running the test with `--no-clean` and manually validating should be fine for now (just make sure to add a comment to the test with the manual steps).

https://github.com/aws/aws-cdk/blob/4cad2cf7e1ca41dedae6adc8866792e5f71b2123/packages/@aws-cdk/aws-events-targets/test/logs/integ.log-group.ts#L77-L85